docs/resources/rules/assertions.md

```diff
@@ -0,0 +1,487 @@
+casing | `string` | Asserts a casing style from this possible list: `camelCase`, `kebab-case`, `snake_case`, `PascalCase`, `MACRO_CASE`, `COBOL-CASE`, `flatcase`. See [casing example](#casing-example).
+mutuallyExclusive | [`string`] | Asserts list of properties (key names only) are mutually exclusive. Use `$keys` in the `on` when this is enabled (for example, `Operation.$keys`). See [mutuallyExclusive example](#mutuallyexclusive-example).
+mutuallyRequired | [`string`] | Asserts list of properties (key names only) are mutually required. Use `$keys` in the `on` when this is enabled (for example, `Operation.$keys`). See [mutuallyRequired example](#mutuallyrequired-example).
+required | [`string`] | Asserts all listed values are defined. See [required example](#required-example).
+disallowed | [`string`] | Asserts all listed values are not defined. See [disallowed example](#disallowed-example).
+defined | `boolean` | Asserts a property is defined. See [defined example](#defined-example).
+undefined | `boolean` | Asserts a property is undefined. See [undefined example](#undefined-example).
+nonEmpty | `boolean` | Asserts a property is not empty. See [nonEmpty example](#nonempty-example).
+minLength | `integer` | Asserts a minimum length (inclusive) of a string or list (array). See [minLength example](#minlength-example).
+maxLength | `integer` | Asserts a maximum length (exclusive) of a string or list (array). See [maxLength example](#maxlength-example).
+sortOrder | `string` \| [Sort Order object](#sort-order-object) | Asserts a sort order to a list (array) of strings or a list of objects by one of the object's property values. See [sortOrder example](#sortorder-example).
+
+## Context object
+
+Property | Type | Description
+-- | -- | --
+type | `string` | **REQUIRED.** One of the [OpenAPI node types](#openapi-node-types).
+filterKeys | [`string`] | The list of key names to continue evaluation with respect to the subject.
+
+See the [context example](#context-example).
+
+## Sort Order object
+
+Property | Type | Description
+-- | -- | --
+direction | `string` | **REQUIRED.** Asserts values are sorted in ascending or descending order using one of these possible values: `asc` or `desc`.
+property | `string` | **REQUIRED.** The name of the property in the collection of objects that will hold values to assess the sort order.
+
+## Examples
+
+The following example shows two assertions each multiple asserts (`defined`, `minLength`, `maxLength`, `pattern`).
+
+The `Operation`, `Tag`, and `Info` properties must:
+- be defined
+- have at least 30 characters
+- end with a _full stop_.
+
+In addition, the `Operation` summary property must:
+- be defined
+- be between 20 and 60 characters
+- not end with a _full stop_.
+
+The following shows how to write that configuration:
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: Tag
+        property: description
```

Comment:
Agree that the key name should be `properties` and it should be an array, so we can evaluate on more that one property from `subject` Node type. But how about nested properties? Is it possible case when we have prop1.prop2 in some Node type?

Reply:
As Adam mentioned above better to keep it flat to simplify docs/usage.

I'm not sure about nested properties. We can technically support it but I'm afraid people may think it will traverse through $ref's... I would keep it simple first and then make it more complex if needed.

Reply:
No nesting.  I'm okay to make it properties like: `string` | [`string`] type. I think it could be helpful.

Reply:
I noticed it was already defined like `string | [string]`. I went to rename it to properties, and then as I was updating the examples I become uncertain if `properties` is a better name than `property` and decided to stick with `property`. I have a small concern that `properties` is overloaded and may be confusing in the documentation between OpenAPI/JSON Schema and that should dominate compared to our "assertions".